### PR TITLE
expression: implement vectorized evaluation for builtinPowSig

### PR DESCRIPTION
**Pull request by tsthght**
### What problem does this PR solve? 
implement vectorized evaluation for builtinPowSig, for #12105

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinMathFunc/builtinPowSig-VecBuiltinFunc-8               	   50000	     30863 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinPowSig-NonVecBuiltinFunc-8            	   30000	     49697 ns/op	       0 B/op	       0 allocs/op
```
### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_math_vec_test.go [hunk 1]
```diff
@@ -49,6 +49,10 @@ var vecBuiltinMathCases = map[string][]vecExprBenchCase{
 	ast.Round: {
 		{types.ETDecimal, []types.EvalType{types.ETDecimal}, nil},
 	},
+	ast.Pow: {
+		{types.ETReal, []types.EvalType{types.ETReal, types.ETReal}, []dataGenerator{&rangeRealGener{0, 10, 0.5}, &rangeRealGener{0, 100, 0.5}}},
+
```
Comment (qw4990) [inline expression/builtin_math_vec_test.go]: This blank line can be removed.

File: expression/builtin_math_vec.go [hunk 2]
```diff
@@ -223,3 +225,47 @@ func (b *builtinRoundDecSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Co
 func (b *builtinRoundDecSig) vectorized() bool {
 	return true
 }
+
+func (b *builtinPowSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf1, err := b.bufAllocator.get(types.ETReal, n)
```
Comment (zz-jason) [inline expression/builtin_math_vec.go]: we can reuse `result` to store one of the results of the two parameters, thus the vectors involved in the calculation can be reduced from 3 to 2. Maybe the performance can be improved by this tiny optimization.

File: expression/builtin_math_vec.go [hunk 3]
```diff
@@ -223,3 +225,47 @@ func (b *builtinRoundDecSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Co
...
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf1); err != nil {
```
Comment (Reminiscent) [inline expression/builtin_math_vec.go]: We can directly use the `result` variable to store the result of args[0]. Thus we can improve the performance by saving a memory copy.

**Comment by zz-jason** [comment]
@tsthght Could you update the benchmark result in the PR description?

File: expression/builtin_math_vec.go [hunk 4]
```diff
@@ -223,3 +225,42 @@ func (b *builtinRoundDecSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Co
...
+
+	if err := b.args[1].VecEvalReal(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	x := buf1.Float64s()
+	y := result.Float64s()
+	f64s := result.Float64s()
+	for i := 0; i < n; i++ {
+		if buf1.IsNull(i) || result.IsNull(i) {
```
Comment (zz-jason) [inline expression/builtin_math_vec.go]: We can use `result.MergeNulls()` to calculate the null bitmap for `result`. It's much faster.

File: expression/builtin_math_vec.go [hunk 5]
```diff
@@ -14,10 +14,11 @@
 package expression
 
 import (
-	"math"
-
+	"fmt"
```
Comment (qw4990) [inline expression/builtin_math_vec.go]: Please format these statements.

**Pull request merged**